x/crosschain/keeper/gas_payment.go

```diff
@@ -128,7 +128,8 @@ func (k Keeper) PayGasNativeAndUpdateCctx(
 	}
 
 	// calculate the final gas fee
-	outTxGasFee := gas.GasLimit.Mul(gas.GasPrice).Add(gas.ProtocolFlatFee)
+	gasFee := gas.GasLimit.Mul(gas.GasPrice)
+	outTxGasFee := gasFee.Add(gas.ProtocolFlatFee)
```

Comment:
can we add comment why is protocol fee added here, same as there is comment above why it is not?